.github/CONTRIBUTING.md

```diff
@@ -210,6 +210,84 @@ cmake -S pybind11/ -B build
 cmake --build build
 ```
 
+### Explanation of the SDist/wheel building design
+
+> These details are _only_ for building from the Python sources from git. The
+> SDists and wheels created do not have any extra requirements at all and are
+> completely normal.
+
+In order to support CMake output files (`pybind11Config.cmake` and helper
```

Comment:
Would reword with an even higher-level picture: "The main objective of the packaging system is to create `sdist` and `wheel` packages that include everything that is needed to work with pybind11, and which can be installed without any additional dependencies. This is more complex than it appears: they must include [..]`

Basically some of the (very useful!) motivational explanations that you and @ktbarrett provided on the PR discussion could be added here in slightly re-phrased form.

Reply:
Actually, this whole paragraph is so full with build system jargon that I'm wondering whether we can really assume pybind11 contributors to be intimately familiar with the subtleties of things like sdists, wheels, manylinux1, [globals].,

Reply:
Pybind11 contributors _who want to work on the packaging_ should be familiar with Python packaging first, I would expect... (Working on reworking this, but maybe proper caveats that most people don't need to know this might help, it's really a technical dive into how this is structured for someone who wants to work on the packaging system)